src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java

```diff
@@ -471,20 +468,33 @@ private String getScoreKeyAndCheckModelAndReadsHarmony() {
         }
     }
 
-    private void setArchitectureAndWeightsFromResources() {
-        if (tensorType.equals(TensorType.read_tensor)) {
-            architecture = IOUtils.writeTempResourceFromPath(resourcePathReadTensor, null).getAbsolutePath();
-            weights = IOUtils.writeTempResourceFromPath(
-                    resourcePathReadTensor.replace(".json", ".hd5"),
-                    null).getAbsolutePath();
-        } else if (tensorType.equals(TensorType.reference)) {
-            architecture = IOUtils.writeTempResourceFromPath(resourcePathReferenceTensor, null).getAbsolutePath();
-            weights = IOUtils.writeTempResourceFromPath(
-                     resourcePathReferenceTensor.replace(".json", ".hd5"), null).getAbsolutePath();
+    private void initializePythonArgsAndModel(){
+        if (weights == null && architecture == null) {
+            if (tensorType.equals(TensorType.read_tensor)) {
+                architecture = IOUtils.writeTempResourceFromPath(resourcePathReadTensor, null).getAbsolutePath();
+                weights = IOUtils.writeTempResourceFromPath(
+                        resourcePathReadTensor.replace(".json", ".hd5"),
+                        null).getAbsolutePath();
+            } else if (tensorType.equals(TensorType.reference)) {
+                architecture = IOUtils.writeTempResourceFromPath(resourcePathReferenceTensor, null).getAbsolutePath();
+                weights = IOUtils.writeTempResourceFromPath(
+                        resourcePathReferenceTensor.replace(".json", ".hd5"), null).getAbsolutePath();
+            } else {
+                throw new GATKException("No default architecture for tensor type:" + tensorType.name());
+            }
+        }
+
+        String getArgsAndModel;
+        if (weights != null && architecture != null) {
```

Comment:
I agree, but can we save this for the PEP8 PR that is on-deck? It will require several python changes to the python code as well as updates to the WDLs, etc so I would rather only refactor that stuff once.